streams/src/main/java/org/apache/kafka/streams/processor/internals/InternalTopologyBuilder.java

```diff
@@ -2067,20 +2073,24 @@ private boolean hasSubscriptionUpdates() {
         return !subscriptionUpdates.isEmpty();
     }
 
-    synchronized void addSubscribedTopicsFromAssignment(final List<TopicPartition> partitions, final String logPrefix) {
+    /**
+     * @return the set of topics that were newly
+     */
+    synchronized Set<String> addSubscribedTopicsFromAssignment(final Set<String> topics, final String logPrefix) {
         if (usesPatternSubscription()) {
```

Comment:
I am a little concerned with this. Are pattern subscriptions the only way to get new topics? what about a namedtopolgy that hasn't been added to this client yet? will this just mean the client will ignore those tasks?

Reply:
We don't want/need to add topics from a named topology (I mean "modular topology" 😉 ) to the subscription if it hasn't been added to the client yet, since it won't be able to process or even correctly direct any records it consumes from those topics until it knows about the actual topology. It doesn't necessarily hurt, the thread would just end up prematurely consuming records for that topic causing them to "waste" memory until the topology is added.

Reply:
But anyways, the main fix in this PR is to make sure we only update each InternalTopologyBuilder with the topics that particular topology is subscribed to, so we shouldn't be invoking this method on topics that don't yet have a corresponding modular topology to begin with